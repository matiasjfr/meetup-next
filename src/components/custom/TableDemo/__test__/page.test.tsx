import { EStatus } from '@prisma/client';
import { render, RenderOptions } from '@testing-library/react';

import { TableDemo } from '..';

const productsMock = [
  {
    id: 1,
    name: 'Product 1',
    description: '',
    buyPrice: null,
    sellPrice: null,
    status: EStatus.ACTIVE,
    image: 'https://loremflickr.com/100/100',
    idCategory: 1,
    category: {
      id: 1,
      name: 'Computers',
      description: 'Outdoors',
      status: EStatus.INACTIVE,
    },
  },
];

describe('TableDemo', () => {
  it('TableDemo correctly', () => {
    const { container } = render(<TableDemo products={productsMock} />);
    expect(container).toMatchSnapshot();
  });
});
