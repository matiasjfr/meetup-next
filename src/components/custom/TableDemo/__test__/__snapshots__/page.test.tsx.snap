// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableDemo TableDemo correctly 1`] = `
<div>
  <div
    class="w-full overflow-auto"
  >
    <table
      class="w-full caption-bottom text-sm"
    >
      <caption
        class="mt-4 text-sm text-muted-foreground"
      >
        A list of your products
      </caption>
      <thead
        class="[&_tr]:border-b"
      >
        <tr
          class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
        >
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 w-[100px]"
          >
            Image
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 w-[20px]"
          >
            ID
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Name
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Status
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Description
          </th>
          <th
            class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
          >
            Category
          </th>
          <th
            class="h-12 px-4 align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 text-center"
          >
            Buy Price
          </th>
          <th
            class="h-12 px-4 align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0 text-center"
          >
            Sell Price
          </th>
        </tr>
      </thead>
      <tbody
        class="[&_tr:last-child]:border-0"
      />
    </table>
  </div>
</div>
`;
